src/Tizen.Applications.Common/Tizen.Applications/AppControl.cs

```diff
@@ -640,6 +638,99 @@ public static void SendTerminateRequest(AppControl terminateRequest)
+                {
+                    appControlResult = (AppControlResult)result;
+                };
+
+                if (replyAfterLaunching != null)
+                {
+                    int requestId = 0;
+                    lock (s_replyNativeCallbackMaps)
+                    {
+                        requestId = s_replyNativeCallbackId++;
+                        s_replyNativeCallbackMaps[requestId] = replyAfterLaunching;
+
+                        s_replyNativeCallback = (launchRequestHandle, replyRequestHandle, result, userData) =>
+                        {
+                            int key = (int)userData;
+                            if (s_replyNativeCallbackMaps.ContainsKey(key))
+                            {
+                                s_replyNativeCallbackMaps[key](new AppControl(launchRequestHandle), new AppControl(replyRequestHandle), (AppControlReplyResult)result);
+                                s_replyNativeCallbackMaps.Remove(key);
+                            }
+                        };
+                    }
+
+                    err = Interop.AppControl.SendLaunchRequestAsync(launchRequest.SafeAppControlHandle, cb, s_replyNativeCallback, (IntPtr)requestId);
+                }
+                else
+                {
+                    err = Interop.AppControl.SendLaunchRequestAsync(launchRequest.SafeAppControlHandle, cb, null, IntPtr.Zero);
+                }
+
+                if (err != Interop.AppControl.ErrorCode.None)
+                {
+                    switch (err)
+                    {
+                        case Interop.AppControl.ErrorCode.InvalidParameter:
+                            throw new ArgumentException("Invalid Arguments");
+                        case Interop.AppControl.ErrorCode.TimedOut:
+                            throw new TimeoutException("Timed out");
+                        case Interop.AppControl.ErrorCode.OutOfMemory:
+                            throw new Exceptions.OutOfMemoryException("Out-of-memory");
+                        case Interop.AppControl.ErrorCode.AppNotFound:
+                            throw new Exceptions.AppNotFoundException("App not found");
+                        case Interop.AppControl.ErrorCode.LaunchRejected:
+                            throw new Exceptions.LaunchRejectedException("Launch rejected");
+                        case Interop.AppControl.ErrorCode.LaunchFailed:
+                            throw new Exceptions.LaunchFailedException("Launch failed");
+                        case Interop.AppControl.ErrorCode.PermissionDenied:
+                            throw new Exceptions.PermissionDeniedException("Permission denied");
+
+                        default:
+                            throw new Exceptions.LaunchRejectedException("Launch rejected");
+                    }
+                }
+
+                return appControlResult;
```

Comment:
It will always returing AppControlResult.None

Reply:
It's not. The method can return None or AppNotFoundException or LaunchFailed. I'll adjust return values.

Reply:
As i know ResultCallback is working as asynchronously
and you do not wait ResultCallback